### PR TITLE
[AMORO-2441] Fix `TableEntriesScan` when no file format is specified in the file name

### LINKED ISSUE #2441: [Bug]: Fix `File format is required` when scanning file entries if no file format is specified in the file name
### What happened?

When there is no file format indicated in the file name(like `xxxx.parquet`, `yyyy.avro`), TableEntriesScan will throw an error `File format is required.`

### Affects Versions

master

### What engines are you seeing the problem on?

Core

### How to reproduce

_No response_

### Relevant log output

```shell
[com.netease.arctic.server.table.executor.SnapshotsExpiringExecutor] [] - unexpected expire error of table arcti
c_xxx.yyy.zzz(tableId=155)
java.lang.IllegalArgumentException: File format is required
        at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:145) ~[iceberg-bundled-guava-1.3.0.jar:?]
        at org.apache.iceberg.DataFiles$Builder.build(DataFiles.java:293) ~[iceberg-core-1.3.0.jar:?]
        at com.netease.arctic.scan.TableEntriesScan.buildDataFile(TableEntriesScan.java:325) ~[amoro-core-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.scan.TableEntriesScan.buildContentFile(TableEntriesScan.java:301) ~[amoro-core-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.scan.TableEntriesScan.lambda$entries$0(TableEntriesScan.java:240) ~[amoro-core-0.6.0-SNAPSHOT.jar:?]
        at org.apache.iceberg.io.CloseableIterable$7$1.next(CloseableIterable.java:202) ~[iceberg-api-1.3.0.jar:?]
        at org.apache.iceberg.io.FilterIterator.advance(FilterIterator.java:65) ~[iceberg-api-1.3.0.jar:?]
        at org.apache.iceberg.io.FilterIterator.hasNext(FilterIterator.java:49) ~[iceberg-api-1.3.0.jar:?]
        at com.netease.arctic.server.utils.IcebergTableUtil.getAllContentFilePath(IcebergTableUtil.java:155) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.MixedTableMaintainer.<init>(MixedTableMaintainer.java:91) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.TableMaintainer.ofTable(TableMaintainer.java:46) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.SnapshotsExpiringExecutor.execute(SnapshotsExpiringExecutor.java:59) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.executeTask(BaseTableExecutor.java:72) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.lambda$scheduleIfNecessary$4(BaseTableExecutor.java:81) ~[amoro-ams-server-0.6.0-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_101]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_101]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_101]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_101]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_101]
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

Close #2441 .

## Brief change log

- fix TableEntriesScan without format suffix

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java [hunk 1]
```diff
@@ -309,11 +310,13 @@ private DataFile buildDataFile(StructLike fileRecord) {
...
     Long fileSize = fileRecord.get(dataFileFieldIndex(DataFile.FILE_SIZE.name()), Long.class);
     Long recordCount = fileRecord.get(dataFileFieldIndex(DataFile.RECORD_COUNT.name()), Long.class);
+    FileFormat format = FileFormat.fromFileName(filePath);
     DataFiles.Builder builder =
         DataFiles.builder(table.spec())
             .withPath(filePath)
             .withFileSizeInBytes(fileSize)
-            .withRecordCount(recordCount);
+            .withRecordCount(recordCount)
+            .withFormat(format != null ? format : FileFormat.PARQUET);
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java:319]: We cannot assume the default table format is `PARQUET`, maybe we can find the file format in the record with filed named `file_format`.

Comment (wangtaohz) [reply core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java:319]: Make sense to me.

**Review by zhoujinsong — APPROVED** [review]
LGTM.
